### PR TITLE
Hide footnotes

### LINKED ISSUE #322: Contracts Table: shows "1 Available to CA agencies only" footnote on all contracts; pls. remove from 2 new Product Categories
When applying a Product Category "sort" to [Contracts Table](https://www.camobilitymarketplace.org/contracts/view), Gillian would like the "1. Available to California agencies only." _footnote_ **removed** from these two Product Category sorts:
• [**Transit Processor**](https://www.camobilitymarketplace.org/contracts/view?contracts-filter-product=Transit%20Processor)
• [**Payment Acceptance Devices**](https://www.camobilitymarketplace.org/contracts/view?contracts-filter-product=Payment%20Acceptance%20Devices)

Images from Gillian to illustrate what we need to remove.
State of California Contracts  California Mobility Marketplace
State of California Contracts  California Mobility Marketplace

**PLEASE NOTE:** Footnote is accurate when it appears under the _entire table_ of contracts or when it appears under the _[Payment Processor](https://www.camobilitymarketplace.org/contracts/view?contracts-filter-product=Payment%20Processor)_ Product Category sort.

Thanks!

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #322

image

image

image

### CODE REVIEW

File: contracts/view.html [hunk 1]
```diff
@@ -92,8 +93,25 @@ <h1 class="page-header">State of California Contracts</h1>
...
+          {% assign products = "" | split: "" %}
+          {% assign footnote_id_str = footnote[0] | append: "" %}
+          
+          {% comment%}
+          Build up product data-attribute string for each footnote 
+          {% endcomment %}
+          {% for entry in site.data.contracts.entries %}
+            {% assign entry_footnotes = entry.footnotes | join: "," | split: "," %}
+            {% if entry_footnotes contains footnote_id_str %}
+              {% unless products contains entry.product %}
```
Comment (thekaveman) [inline contracts/view.html:105]: So this `{% unless %}` guards against adding the same product twice, when multiple product entries have the same footnote?

File: scripts/contracts/footer.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const updateFooter = filters => {
+  const productFilters = filters.filter(f => f.field === 'product' && f.type === 'like');
+  
+  const footnotes = $('footer').children('ol').children('li');
+  footnotes.each((i, elem) => {
+    const _dataAttr = $(elem).attr("data-products");
+    
+    const footnoteProducts = _dataAttr ? _dataAttr.split(",") : [];
+    const selectedProducts = new Set(productFilters.map(f => f.value));
+    const intersection = new Set(footnoteProducts.filter(p => selectedProducts.has(p)));
```
Comment (thekaveman) [inline scripts/contracts/footer.js:15]: Very unfortunate there is no built in `Set` intersection operator!

File: scripts/smart-table.js [hunk 3]
```diff
@@ -92,6 +96,8 @@ $(function () {
           pillContainer.prepend($pill[0]);
           pillFilters.add(cellValue);
         }
+
+        smartTableCallbacks.dataFiltered.forEach(cb => cb(filters));
```
Comment (thekaveman) [inline scripts/smart-table.js:100]: Using a list of callbacks here is really clever! Gives us a nice way to expand this in the future if the need arises.

File: scripts/contracts/footer.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+/*
+  `updateFooter` is specific to the Contracts table in `contracts/view.html`.
+  
+  It integrates with `smart-table.js` to be called when data is filtered.
+*/
+const updateFooter = filters => {
+  const productFilters = filters.filter(f => f.field === 'product' && f.type === 'like');
+  
+  const footnotes = $('footer').children('ol').children('li');
```
Comment (thekaveman) [inline scripts/contracts/footer.js:9]: I think my only requested change would be this line - the `$('footer')` selector seems too broad (we have 2 `<footer>` elements on the page already).

Can we use the footnotes list selector instead? `ol.footnote-listing`. That way we're sure to get the list of footnote items.

**Review by thekaveman — CHANGES_REQUESTED** [review]
Just one minor change requested. 

Otherwise this is super clever @angela-tran! I love the implementation 🚀

File: scripts/contracts/footer.js (hunk 4 above)
Comment (angela-tran) [reply scripts/contracts/footer.js]: Good suggestion 😄 - I changed it to `$('ol.footnote-listing').children('li')`

File: contracts/view.html (hunk 1 above)
Comment (angela-tran) [reply contracts/view.html:105]: Yeah, unfortunate that they don't have a way of doing `not contains`.

File: scripts/contracts/footer.js (hunk 4 above)
Comment (angela-tran) [reply scripts/contracts/footer.js]: https://github.com/cal-itp/mobility-marketplace/pull/325/files#diff-4b3572daf35019e63906407ff17148ff60f18b7cb4b0825555c1131e3ed40b5fR9

**Review by thekaveman — APPROVED** [review]
Nitpick: We should have a consistent quote character in JS files, we're mixing single and double quotes in the `scripts/contracts/footer.js` file.

(My preference is always double)

**Comment by angela-tran** [comment]
> Nitpick: We should have a consistent quote character in JS files, we're mixing single and double quotes in the `scripts/contracts/footer.js` file.
> 
> (My preference is always double)

Thanks for pointing this out - I will go ahead and make it consistent here!

**Pull request merged**